### PR TITLE
Guard admin routes

### PR DESCRIPTION
**Pull request by kevinxiao27**
#### Changes: 
- Added package @amplify/adapter-nextjs
- Added amplify-utils.ts (enables amplify api calls on server side, required for next.js)
- Added ConfigureAmplify.tsx, ensures same instance is run for client side with next.js
- Added middleware.ts to guard admin paths, calls on runWithAmplifyServerContext that checks for user email ending with biztech.com domain to determine isAdmin
- moved amplify config to layout.tsx

#### Testing: 
[video link](https://drive.google.com/file/d/1StFJG-c2g6jcQROOAShhjFJqpPRnJ99M/view?usp=sharing)
(since oauth isnt implemented in the current login component I changed the route guard to check for the gmail.com domain using my biztech account from last year)

#### Out of Scope:
- login component has styling issues (background appears as white from what is currently merged in the dev branch)

For Issue: #28 

Note:
- follows this guide from amplify [link](https://docs.amplify.aws/gen1/javascript/build-a-backend/server-side-rendering/nextjs/)
- only paths under the matcher will have this middleware run, if used for anything else this guard might need to be rewritten
- Only relevant changes to login.tsx is that the amplify.config was moved to layout. Not sure why my prettier does that but i'll try to fix it so huge diffs don't appear because of formatting.

### CODE REVIEW

**Comment by AllanT102** [comment]
Can you add a video for test evidence? @kevinxiao27

**Comment by kevinxiao27** [comment]
> Can you add a video for test evidence? @kevinxiao27

added to PR

File: src/components/ConfigureAmplify.tsx
Comment (kevinxiao27) [inline src/components/ConfigureAmplify.tsx:1]: component that runs on root level to for amplify context on client side

File: src/pages/login.tsx
Comment (kevinxiao27) [inline src/pages/login.tsx:1]: no major changes except that Amplify.configure is now handled by the configureAmplifyClientSide component. my prettier is wack i'll try to get it to match existing codebase so for the future huge diffs like this don't appear for no reason

File: src/util/amplify-utils.ts
Comment (kevinxiao27) [inline src/util/amplify-utils.ts:1]: needed for next.js because middleware runs on server side

**Review by AllanT102 — APPROVED** [review]

**Pull request merged**